### PR TITLE
Abstract ups

### PR DESCRIPTION
**Pull request by Karthi8883**

### CODE REVIEW

File: ontology/yaml/resources/fields/telemetry_fields.yaml [hunk 1]
```diff
@@ -1382,12 +1382,45 @@ literals:
...
+- bypass_status:
+  - ACTIVE
+  - INACTIVE
+- inverter_mode:
+  - SYNCHRONOUS
+  - ASYNCHRONOUS
+- overload_alarm:
+  - ACTIVE
+  - INACTIVE
+- bypass_phase_status:
```
Comment (tasodorff) [inline ontology/yaml/resources/fields/telemetry_fields.yaml:1398]: what is this for? not understanding from the field name what its usage should be.

Comment (tasodorff) [reply ontology/yaml/resources/fields/telemetry_fields.yaml:1398]: im okay with this.

File: ontology/yaml/resources/states/states.yaml [hunk 2]
```diff
@@ -69,5 +69,10 @@ POSITIONA: "Automatic Transfer switch position 0"
 POSITIONB: "Automatic Transfer switch position 1"
 POSITIONC: "Automatic Transfer switch position 2"
 
-
-
+#FOR UPS
+BATTERY: "Operation mode or input power is via Battery"
+SYNCHRONOUS: "Mode of operation of inverter"
+ASYNCHRONOUS: "Mode of operation of inverter"
```
Comment (tasodorff) [inline ontology/yaml/resources/states/states.yaml:75]: please differentiate between lines 74 and 75, they have the same definition: what do synchronous and asynchronous mean?

File: ontology/yaml/resources/states/states.yaml [hunk 3]
```diff
@@ -69,5 +69,10 @@ POSITIONA: "Automatic Transfer switch position 0"
...
+SOFTSTART: "Mode of operation of rectifier"
```
Comment (tasodorff) [inline ontology/yaml/resources/states/states.yaml:76]: Same. describe the operation

**Review by tasodorff — CHANGES_REQUESTED** [review]
please add defs and we are good to go

**Comment by tasodorff** [comment]
@Karthi8883 please revise definitions to describe explicitly what each state means and we are good to approve/merge.

File: ontology/yaml/resources/states/states.yaml (hunk 2 above)
Comment (Karthi8883) [reply ontology/yaml/resources/states/states.yaml]: Added the definition in detail.

File: ontology/yaml/resources/states/states.yaml (hunk 3 above)
Comment (Karthi8883) [reply ontology/yaml/resources/states/states.yaml]: Added the definition in detail

**Comment by Karthi8883** [comment]
> @Karthi8883 please revise definitions to describe explicitly what each state means and we are good to approve/merge.

@tasodorff  revised definitions with explanation

**Review by tasodorff — APPROVED** [review]

**Comment by mschulze17** [comment]
@cstirdivant yours to merge

**Pull request merged**